wp-web-push/lib/delapuente/wp-sw-manager/class-wp-sw-manager.php

```diff
@@ -0,0 +1,179 @@
+     *
+     * @see WP_SW_Manager::get_js_id()
+     * @api
+     * @var string
+     */
+    const SW_REGISTRAR_SCRIPT = 'wp-sw-manager-registrar';
+
+    const SW_REGISTRAR_SCRIPT_URL = 'wpswmanager/sw-registrar.js';
+
+    private static $instance;
+
+    /**
+     * Obtains the shared manager.
+     *
+     * @api
+     * @returns WP_SW_Manager The shared manager instance.
+     */
+    public static function get_manager() {
+        if (!self::$instance) {
+            self::$instance = new self();
+        }
+        return self::$instance;
+    }
+
+    private $router;
+
+    private $service_workers;
+
+    private function __construct() {
+        $this->router = WP_SW_Manager_Router::get_router();
+        $this->service_workers = array();
+        $this->setup_sw_registrar_script();
+    }
+
+    /**
+     * Selects the combinator representing the service worker to write into.
+     *
+     * As only one service worker per scope is currently allowed by the
+     * ServiceWorker API, the manager identify the proper service worker
+     * based on the scope only.
+     *
+     * @api
+     * @param string $scope The scope used to select the service worker.
+     * If omitted, it defaults to `home_url('/', 'relative')`;
+     * @return WP_SW_Combinator The combinator instance to generate the
+     * content of the service worker.
+     */
+    public function sw($scope='') {
+        if (!$scope) { $scope = $this->default_scope(); }
+        if (!array_key_exists($scope, $this->service_workers)) {
+            $this->add_new_sw($scope);
+        }
+        return $this->service_workers[$scope];
+    }
+
+    /**
+     * Obtains the JavaScript ID to select the registration promise in
+     * JavaScript client code.
+     *
+     * When attaching content to a service worker combinator, the worker
+     * is automatically [registered](https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerContainer/register)
+     * with a client JavaScript script. If you need to access this registration,
+     * you can use the `$swRegistrations` object in your client scripts. This
+     * object is a map between service workers and registration promises. To index
+     * the proper service worker you need a unique key that can be retrieved
+     * with this function.
+     *
+     * @api
+     * @param string $scope The scope identifying the service worker.
+     * If omitted, it defaults to `home_url('/', 'relative')`;
+     * @return string A string representing the unique identifier for the
+     * service worker registration promise.
+     */
+    public function sw_js_id($scope='') {
+        if (!$scope) { $scope = $this->default_scope(); }
+        return $scope;
+    }
+
+    private function default_scope() {
+        return home_url('/', 'relative');
+    }
+
+    private function setup_sw_registrar_script() {
+        $this->router->add_route(self::SW_REGISTRAR_SCRIPT_URL,  array($this, 'sw_registrar'));
```

Comment:
Nit: you could store this in a property of the class, so you don't need to call `route_url` in `enqueue_registrar` and you could then remove `route_url` from the Router class.